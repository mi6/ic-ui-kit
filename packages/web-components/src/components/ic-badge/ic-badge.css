@media (prefers-reduced-motion: no-preference) {
  :host(.ic-badge-show) {
    animation: expand var(--ic-transition-duration-slow);
  }

  :host(.ic-badge-hide) {
    animation: shrink var(--ic-transition-duration-slow);
  }
}

:host {
  display: flex;
  height: var(--ic-space-md);
  min-width: var(--ic-space-md);
  width: fit-content;
  border-radius: calc(2 * var(--ic-space-xxl));
  position: absolute;
}

:host(.ic-badge-neutral) {
  background-color: var(--ic-architectural-500);
}

:host(.ic-badge-light) {
  background-color: var(--ic-architectural-40);
}

:host(.ic-badge-info) {
  background-color: var(--ic-status-info);
}

:host(.ic-badge-warning) {
  background-color: var(--ic-status-warning);
}

:host(.ic-badge-error) {
  background-color: var(--ic-status-error);
}

:host(.ic-badge-success) {
  background-color: var(--ic-status-success);
}

:host(.ic-badge-small) {
  height: var(--ic-space-sm);
  min-width: var(--ic-space-sm);
}

:host(.ic-badge-large) {
  height: calc(var(--ic-space-md) + var(--ic-space-xxs));
  min-width: calc(var(--ic-space-md) + var(--ic-space-xxs));
}

:host(.ic-badge-dot.ic-badge-default) {
  height: var(--ic-space-xs);
  width: var(--ic-space-xs);
  min-width: var(--ic-space-xs);
}

:host(.ic-badge-dot.ic-badge-small) {
  height: calc(var(--ic-space-xxs) + var(--ic-space-xxxs));
  width: calc(var(--ic-space-xxs) + var(--ic-space-xxxs));
  min-width: calc(var(--ic-space-xxs) + var(--ic-space-xxxs));
}

:host(.ic-badge-dot.ic-badge-large) {
  height: var(--ic-space-sm);
  width: var(--ic-space-sm);
  min-width: var(--ic-space-sm);
}

:host(.ic-badge-foreground-dark) ::slotted(*) {
  fill: var(--ic-color-primary-text);
}

:host(.ic-badge-foreground-light) ::slotted(*) {
  fill: white;
}

:host(.ic-badge-foreground-dark) ic-typography {
  color: var(--ic-color-primary-text);
}

:host(.ic-badge-foreground-light) ic-typography {
  color: white;
}

:host(.ic-badge-text) ic-typography {
  align-self: center;
  padding: 0 calc((var(--ic-space-xs) + var(--ic-space-1px)) / 2)
    var(--ic-space-1px);
}

:host(.ic-badge-text.ic-badge-small) ic-typography {
  padding: 0 0.2132rem;
}

:host(.ic-badge-text.ic-badge-large) ic-typography {
  padding: 0 calc((var(--ic-space-sm) + var(--ic-space-1px)) / 2)
    var(--ic-space-1px);
}

:host(.ic-badge-icon) ::slotted(svg) {
  width: var(--ic-space-sm);
  height: var(--ic-space-sm);
  padding: var(--ic-space-xxxs);
}

:host(.ic-badge-icon.ic-badge-small) ::slotted(svg) {
  width: var(--ic-space-xs);
  height: var(--ic-space-xs);
}

:host(.ic-badge-icon.ic-badge-large) ::slotted(svg) {
  width: calc(var(--ic-space-sm) + var(--ic-space-xxxs));
  height: calc(var(--ic-space-sm) + var(--ic-space-xxxs));
  padding: calc(var(--ic-space-xxxs) + var(--ic-space-1px));
}

:host(.ic-badge-far) {
  top: calc(-1 * var(--ic-space-xs));
  right: calc(-1 * var(--ic-space-xs));
}

:host(.ic-badge-far.ic-badge-small),
:host(.ic-badge-dot.ic-badge-far.ic-badge-large) {
  top: calc(-1 * var(--ic-space-xxs));
  right: calc(-1 * var(--ic-space-xxs));
}

:host(.ic-badge-dot.ic-badge-far),
:host(.ic-badge-dot.ic-badge-far.ic-badge-small) {
  top: calc(-1 * var(--ic-space-xxxs));
  right: calc(-1 * var(--ic-space-xxxs));
}

:host(.ic-badge-near) {
  top: calc(-1 * calc(var(--ic-space-xxs) + var(--ic-space-1px)));
  right: calc(-1 * calc(var(--ic-space-xxs) + var(--ic-space-1px)));
}

:host(.ic-badge-dot.ic-badge-near) {
  top: calc(-1 * var(--ic-space-1px));
  right: calc(-1 * var(--ic-space-1px));
}

:host(.ic-badge-inline) {
  position: static;
}

:host(.ic-badge-hide) {
  visibility: hidden !important;
  transition: visibility var(--ic-transition-duration-slow);
}

.sr-only {
  position: absolute;
  left: -9999px;
}

@keyframes expand {
  from {
    opacity: 0;
    transform: scale(0);
  }

  to {
    opacity: 1;
    transform: scale(1);
  }
}

@keyframes shrink {
  from {
    opacity: 1;
    transform: scale(1);
  }

  to {
    opacity: 0;
    transform: scale(0);
  }
}
