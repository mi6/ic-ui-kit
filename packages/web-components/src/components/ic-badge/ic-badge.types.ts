import { IcStatusVariants } from "../../utils/types";

export type IcBadgeTypes = "dot" | "text" | "icon";

export type IcBadgePositions = "far" | "near" | "inline";

export type IcBadgeVariants = IcStatusVariants | "light" | "custom";
