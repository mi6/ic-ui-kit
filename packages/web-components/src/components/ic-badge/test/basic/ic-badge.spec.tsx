import { newSpecPage } from "@stencil/core/testing";
import { Badge } from "../../ic-badge";
import { Button } from "../../../ic-button/ic-button";
import { Chip } from "../../../ic-chip/ic-chip";
import { Tab } from "../../../ic-tab/ic-tab";
import { Card } from "../../../ic-card/ic-card";
import { NavigationButton } from "../../../ic-navigation-button/ic-navigation-button";
import { NavigationItem } from "../../../ic-navigation-item/ic-navigation-item";

describe("ic-badge", () => {
  it("should render with text slotted in a button", async () => {
    const page = await newSpecPage({
      components: [Button, Badge],
      html: `<ic-button>Button<ic-badge slot="badge" text-label="1"/></ic-button>`,
    });

    expect(page.root).toMatchSnapshot(
      "should render with text slotted in a button"
    );
  });

  it("should render light variant", async () => {
    const page = await newSpecPage({
      components: [Button, Badge],
      html: `<ic-button>Button<ic-badge slot="badge" variant="light" text-label="1"/></ic-button>`,
    });

    expect(page.root).toMatchSnapshot("should render light variant");
  });

  it("should render error variant", async () => {
    const page = await newSpecPage({
      components: [Button, Badge],
      html: `<ic-button>Button<ic-badge slot="badge" variant="error" text-label="1"/></ic-button>`,
    });

    expect(page.root).toMatchSnapshot("should render error variant");
  });

  it("should render success variant", async () => {
    const page = await newSpecPage({
      components: [Button, Badge],
      html: `<ic-button>Button<ic-badge slot="badge" variant="success" text-label="1"/></ic-button>`,
    });

    expect(page.root).toMatchSnapshot("should render success variant");
  });

  it("should render warning variant", async () => {
    const page = await newSpecPage({
      components: [Button, Badge],
      html: `<ic-button>Button<ic-badge slot="badge" variant="warning" text-label="1"/></ic-button>`,
    });

    expect(page.root).toMatchSnapshot("should render warning variant");
  });
  it("should render info variant", async () => {
    const page = await newSpecPage({
      components: [Button, Badge],
      html: `<ic-button>Button<ic-badge slot="badge" variant="info" text-label="1"/></ic-button>`,
    });

    expect(page.root).toMatchSnapshot("should render info variant");
  });

  it("should render custom variant with custom colour in hex", async () => {
    const page = await newSpecPage({
      components: [Button, Badge],
      html: `<ic-button>Button<ic-badge slot="badge" variant="custom" text-label="1" custom-color="#F8C8DC"/></ic-button>`,
    });

    expect(page.root).toMatchSnapshot(
      "should render custom variant with custom colour in hex"
    );
  });

  it("should render custom variant with custom colour in rgb", async () => {
    const page = await newSpecPage({
      components: [Button, Badge],
      html: `<ic-button>Button<ic-badge slot="badge" variant="custom" text-label="1" custom-color="rgb(248,200,220)"/></ic-button>`,
    });

    expect(page.root).toMatchSnapshot(
      "should render custom variant with custom colour in rgb"
    );
  });

  it("should render with max number prop set", async () => {
    const page = await newSpecPage({
      components: [Button, Badge],
      html: `<ic-button>Button<ic-badge slot="badge" text-label="100" max-number="9"/></ic-button>`,
    });

    expect(page.root).toMatchSnapshot("should render with max number prop set");
  });

  it("should render with accessible label", async () => {
    const page = await newSpecPage({
      components: [Button, Badge],
      html: `<ic-button>Button<ic-badge slot="badge" text-label="1" accessible-label="1 notification found"/></ic-button>`,
    });

    expect(page.root).toMatchSnapshot("should render with accessible label");
  });

  it("should render size small", async () => {
    const page = await newSpecPage({
      components: [Button, Badge],
      html: `<ic-button>Button<ic-badge slot="badge" text-label="100" size="small"/></ic-button>`,
    });

    expect(page.root).toMatchSnapshot("should render size small");
  });

  it("should render size large", async () => {
    const page = await newSpecPage({
      components: [Button, Badge],
      html: `<ic-button>Button<ic-badge slot="badge" text-label="100" size="large"/></ic-button>`,
    });

    expect(page.root).toMatchSnapshot("should render size large");
  });

  it("should render type dot", async () => {
    const page = await newSpecPage({
      components: [Button, Badge],
      html: `<ic-button>Button<ic-badge slot="badge" type="dot"/></ic-button>`,
    });

    expect(page.root).toMatchSnapshot("should render type dot");
  });

  it("should render type icon", async () => {
    const page = await newSpecPage({
      components: [Button, Badge],
      html: `<ic-button>Button<ic-badge slot="badge" type="icon">
            <svg slot='badge-icon' xmlns="http://www.w3.org/2000/svg" height="24px" viewBox="0 0 24 24" width="24px" fill="#000000">
            <path d="M0 0h24v24H0V0z" fill="none"/>
            <path d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"/>
            </svg>
            </ic-badge></ic-button>`,
    });

    expect(page.root).toMatchSnapshot("should render type icon");
  });

  it("should render with visible false", async () => {
    const page = await newSpecPage({
      components: [Button, Badge],
      html: `<ic-button>Button<ic-badge slot="badge" visible="false" text-label="1"/></ic-button>`,
    });

    expect(page.root).toMatchSnapshot("should render with visible false");
  });

  it("should render slotted in a chip", async () => {
    const page = await newSpecPage({
      components: [Chip, Badge],
      html: `<ic-chip label="Badge"><ic-badge slot="badge" text-label="1" position="near"/></ic-chip>`,
    });

    expect(page.root).toMatchSnapshot("should render slotted in a chip");
  });

  it("should render slotted in a tab with aria-label set on badge", async () => {
    const page = await newSpecPage({
      components: [Tab, Badge],
      html: `<ic-tab>Tab<ic-badge slot="badge" text-label="1" position="inline"/></ic-tab>`,
    });

    expect(page.root).toMatchSnapshot("should render slotted in a tab");
  });

  it("should render slotted in a card with aria-label set on badge", async () => {
    const page = await newSpecPage({
      components: [Card, Badge],
      html: `<ic-card heading="Badge"><ic-badge slot="badge" text-label="1" position="near"/></ic-card>`,
    });

    expect(page.root).toMatchSnapshot("should render slotted in a card");
  });

  it("should render slotted in a navigation button", async () => {
    const page = await newSpecPage({
      components: [NavigationButton, Badge],
      html: `<ic-navigation-button
      label="Button One"
      slot="buttons"
      href="https://www.google.com"
      target="_blank"
      title="Google 1"
    >
      <svg
        slot="icon"
        xmlns="http://www.w3.org/2000/svg"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        fill="#000000"
      >
        <path d="M0 0h24v24H0V0z" fill="none" />
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5.5-2.5l7.51-3.49L17.5 6.5 9.99 9.99 6.5 17.5zm5.5-6.6c.61 0 1.1.49 1.1 1.1s-.49 1.1-1.1 1.1-1.1-.49-1.1-1.1.49-1.1 1.1-1.1z"
        />
      </svg>
      <ic-badge
        text-label="1"
        slot="badge"
        position="near"
        variant="light"
      ></ic-badge>
    </ic-navigation-button>`,
    });

    expect(page.root).toMatchSnapshot(
      "should render slotted in a navigation button"
    );
  });

  it("should render slotted in a navigation item", async () => {
    const page = await newSpecPage({
      components: [NavigationItem, Badge],
      html: `<ic-navigation-item label="Navigation" href="/">
      <ic-badge
        text-label="1"
        slot="badge"
        variant="light"
        position="far"
      ></ic-badge>
      <svg
        slot="icon"
        width="24"
        height="24"
        viewBox="0 0 24 24"
        fill="none"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M23 8C23 9.1 22.1 10 21 10C20.82 10 20.65 9.98 20.49 9.93L16.93 13.48C16.98 13.64 17 13.82 17 14C17 15.1 16.1 16 15 16C13.9 16 13 15.1 13 14C13 13.82 13.02 13.64 13.07 13.48L10.52 10.93C10.36 10.98 10.18 11 10 11C9.82 11 9.64 10.98 9.48 10.93L4.93 15.49C4.98 15.65 5 15.82 5 16C5 17.1 4.1 18 3 18C1.9 18 1 17.1 1 16C1 14.9 1.9 14 3 14C3.18 14 3.35 14.02 3.51 14.07L8.07 9.52C8.02 9.36 8 9.18 8 9C8 7.9 8.9 7 10 7C11.1 7 12 7.9 12 9C12 9.18 11.98 9.36 11.93 9.52L14.48 12.07C14.64 12.02 14.82 12 15 12C15.18 12 15.36 12.02 15.52 12.07L19.07 8.51C19.02 8.35 19 8.18 19 8C19 6.9 19.9 6 21 6C22.1 6 23 6.9 23 8Z"
          fill="currentColor"
        />
      </svg>
    </ic-navigation-item>`,
    });

    expect(page.root).toMatchSnapshot(
      "should render slotted in a navigation item"
    );
  });

  it("should hide and show the badge using visible prop", async () => {
    const page = await newSpecPage({
      components: [Badge],
      html: `<ic-button>Button<ic-badge slot="badge" text-label="1"/></ic-button>`,
    });

    const badge = document.querySelector("ic-badge");

    expect(page.rootInstance.visible).toBe(true);

    badge.visible = false;
    await page.waitForChanges();

    expect(page.rootInstance.visible).toBe(false);

    badge.visible = true;
    await page.waitForChanges();

    expect(page.rootInstance.visible).toBe(true);
  });

  //To be removed when we remove show and hide methods
  it("should hide and show the badge using methods", async () => {
    const page = await newSpecPage({
      components: [Badge],
      html: `<ic-button>Button<ic-badge slot="badge" text-label="1"/></ic-button>`,
    });

    const badge = document.querySelector("ic-badge");

    expect(page.rootInstance.visible).toBe(true);

    await badge.hideBadge();
    await page.waitForChanges();

    expect(page.rootInstance.visible).toBe(false);

    await badge.showBadge();
    await page.waitForChanges();

    expect(page.rootInstance.visible).toBe(true);
  });

  it("should set the correct badge colour", async () => {
    const page = await newSpecPage({
      components: [Badge],
      html: `<ic-button>Button<ic-badge slot="badge" variant="custom" text-label="1" custom-color="#F8C8DC"/></ic-button>`,
    });

    await page.rootInstance.setBadgeColour();

    expect(page.rootInstance.customColorRGBA).toEqual({
      a: 1,
      b: 220,
      g: 200,
      r: 248,
    });
  });

  it("should set the correct foreground colour", async () => {
    const page = await newSpecPage({
      components: [Badge],
      html: `<ic-button>Button<ic-badge slot="badge" variant="custom" text-label="1" custom-color="#F8C8DC"/></ic-button>`,
    });

    await page.rootInstance.getBadgeForeground();

    expect(page.rootInstance.foregroundColour).toBe("dark");
  });

  it("should set the correct text label when max number is set", async () => {
    const page = await newSpecPage({
      components: [Badge],
      html: `<ic-button>Button<ic-badge slot="badge" text-label="100" max-number="9"/></ic-button>`,
    });

    expect(page.rootInstance.getTextLabel()).toBe("9+");
  });

  it("should render with id set", async () => {
    const page = await newSpecPage({
      components: [Button, Badge],
      html: `<ic-button>Button<ic-badge slot="badge" id="badge-1" text-label="1"/></ic-button>`,
    });

    expect(page.root).toMatchSnapshot("should render with id");
  });
});
