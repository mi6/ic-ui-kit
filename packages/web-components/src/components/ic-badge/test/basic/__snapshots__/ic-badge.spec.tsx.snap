// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-badge should render custom variant with custom colour in hex: should render custom variant with custom colour in hex 1`] = `
<ic-button class="button-size-default button-variant-primary with-badge" exportparts="button">
  <mock:shadow-root>
    <slot name="badge"></slot>
    <button class="button" part="button" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
  <ic-badge aria-label="badge being displayed" class="custom default far foreground-dark show text" custom-color="#F8C8DC" role="status" slot="badge" text-label="1" variant="custom" style="background-color: rgba(248, 200, 220, 1);">
    <mock:shadow-root>
      <ic-typography variant="badge">
        1
      </ic-typography>
    </mock:shadow-root>
  </ic-badge>
</ic-button>
`;

exports[`ic-badge should render custom variant with custom colour in rgb: should render custom variant with custom colour in rgb 1`] = `
<ic-button class="button-size-default button-variant-primary with-badge" exportparts="button">
  <mock:shadow-root>
    <slot name="badge"></slot>
    <button class="button" part="button" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
  <ic-badge aria-label="badge being displayed" class="custom default far foreground-dark show text" custom-color="rgb(248,200,220)" role="status" slot="badge" text-label="1" variant="custom" style="background-color: rgba(248, 200, 220, 1);">
    <mock:shadow-root>
      <ic-typography variant="badge">
        1
      </ic-typography>
    </mock:shadow-root>
  </ic-badge>
</ic-button>
`;

exports[`ic-badge should render error variant: should render error variant 1`] = `
<ic-button class="button-size-default button-variant-primary with-badge" exportparts="button">
  <mock:shadow-root>
    <slot name="badge"></slot>
    <button class="button" part="button" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
  <ic-badge aria-label="badge being displayed" class="default error far foreground-light show text" role="status" slot="badge" text-label="1" variant="error">
    <mock:shadow-root>
      <ic-typography variant="badge">
        1
      </ic-typography>
    </mock:shadow-root>
  </ic-badge>
</ic-button>
`;

exports[`ic-badge should render info variant: should render info variant 1`] = `
<ic-button class="button-size-default button-variant-primary with-badge" exportparts="button">
  <mock:shadow-root>
    <slot name="badge"></slot>
    <button class="button" part="button" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
  <ic-badge aria-label="badge being displayed" class="default far foreground-light info show text" role="status" slot="badge" text-label="1" variant="info">
    <mock:shadow-root>
      <ic-typography variant="badge">
        1
      </ic-typography>
    </mock:shadow-root>
  </ic-badge>
</ic-button>
`;

exports[`ic-badge should render light variant: should render light variant 1`] = `
<ic-button class="button-size-default button-variant-primary with-badge" exportparts="button">
  <mock:shadow-root>
    <slot name="badge"></slot>
    <button class="button" part="button" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
  <ic-badge aria-label="badge being displayed" class="default far foreground-light light show text" role="status" slot="badge" text-label="1" variant="light">
    <mock:shadow-root>
      <ic-typography variant="badge">
        1
      </ic-typography>
    </mock:shadow-root>
  </ic-badge>
</ic-button>
`;

exports[`ic-badge should render size large: should render size large 1`] = `
<ic-button class="button-size-default button-variant-primary with-badge" exportparts="button">
  <mock:shadow-root>
    <slot name="badge"></slot>
    <button class="button" part="button" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
  <ic-badge aria-label="badge being displayed" class="far foreground-light large neutral show text" role="status" size="large" slot="badge" text-label="100">
    <mock:shadow-root>
      <ic-typography variant="badge">
        100
      </ic-typography>
    </mock:shadow-root>
  </ic-badge>
</ic-button>
`;

exports[`ic-badge should render size small: should render size small 1`] = `
<ic-button class="button-size-default button-variant-primary with-badge" exportparts="button">
  <mock:shadow-root>
    <slot name="badge"></slot>
    <button class="button" part="button" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
  <ic-badge aria-label="badge being displayed" class="far foreground-light neutral show small text" role="status" size="small" slot="badge" text-label="100">
    <mock:shadow-root>
      <ic-typography variant="badge-small">
        100
      </ic-typography>
    </mock:shadow-root>
  </ic-badge>
</ic-button>
`;

exports[`ic-badge should render slotted in a card: should render slotted in a card 1`] = `
<ic-card heading="Badge">
  <mock:shadow-root>
    <div class="card">
      <div class="card-header">
        <div class="card-title">
          <slot name="heading">
            <ic-typography variant="h4">
              <p>
                Badge
              </p>
            </ic-typography>
          </slot>
        </div>
      </div>
      <slot name="badge"></slot>
    </div>
  </mock:shadow-root>
  <ic-badge aria-label="badge being displayed" class="default foreground-light near neutral show text" position="near" role="status" slot="badge" text-label="1">
    <mock:shadow-root>
      <ic-typography variant="badge">
        1
      </ic-typography>
    </mock:shadow-root>
  </ic-badge>
</ic-card>
`;

exports[`ic-badge should render slotted in a chip: should render slotted in a chip 1`] = `
<ic-chip label="Badge">
  <mock:shadow-root>
    <div class="chip default filled">
      <ic-typography class="label" variant="label">
        <span>
          Badge
        </span>
      </ic-typography>
      <slot name="badge"></slot>
    </div>
  </mock:shadow-root>
  <ic-badge aria-label="badge being displayed" class="default foreground-light near neutral show text" position="near" role="status" slot="badge" text-label="1">
    <mock:shadow-root>
      <ic-typography variant="badge">
        1
      </ic-typography>
    </mock:shadow-root>
  </ic-badge>
</ic-chip>
`;

exports[`ic-badge should render slotted in a navigation button: should render slotted in a navigation button 1`] = `
<ic-navigation-button href="https://www.google.com" label="Button One" slot="buttons" target="_blank">
  <mock:shadow-root>
    <ic-button appearance="light" aria-label="Button One" href="https://www.google.com" size="large" target="_blank" title="Google 1" variant="icon">
      <slot name="icon" slot="left-icon"></slot>
      <slot name="badge"></slot>
    </ic-button>
  </mock:shadow-root>
  <svg fill="#000000" height="24px" slot="icon" viewBox="0 0 24 24" width="24px" xmlns="http://www.w3.org/2000/svg">
    <path d="M0 0h24v24H0V0z" fill="none"></path>
    <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5.5-2.5l7.51-3.49L17.5 6.5 9.99 9.99 6.5 17.5zm5.5-6.6c.61 0 1.1.49 1.1 1.1s-.49 1.1-1.1 1.1-1.1-.49-1.1-1.1.49-1.1 1.1-1.1z"></path>
  </svg>
  <ic-badge aria-label="badge being displayed" class="default foreground-light light near show text" position="near" role="status" slot="badge" text-label="1" variant="light">
    <mock:shadow-root>
      <ic-typography variant="badge">
        1
      </ic-typography>
    </mock:shadow-root>
  </ic-badge>
</ic-navigation-button>
`;

exports[`ic-badge should render slotted in a navigation item: should render slotted in a navigation item 1`] = `
<ic-navigation-item class="light navigation-item" href="/" label="Navigation" role="listitem">
  <mock:shadow-root>
    <ic-tooltip class="tooltip-navigation-item" label="Navigation" placement="right" target="navigation-item">
      <a class="link" href="/" part="link">
        <div class="icon">
          <slot name="icon"></slot>
          <slot name="badge"></slot>
        </div>
        <ic-typography variant="label">
          Navigation
        </ic-typography>
        <div class="chevron-container"></div>
      </a>
    </ic-tooltip>
  </mock:shadow-root>
  <ic-badge aria-label="badge being displayed" class="default far foreground-light light show text" position="far" role="status" slot="badge" text-label="1" variant="light">
    <mock:shadow-root>
      <ic-typography variant="badge">
        1
      </ic-typography>
    </mock:shadow-root>
  </ic-badge>
  <svg fill="none" height="24" slot="icon" viewBox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg">
    <path d="M23 8C23 9.1 22.1 10 21 10C20.82 10 20.65 9.98 20.49 9.93L16.93 13.48C16.98 13.64 17 13.82 17 14C17 15.1 16.1 16 15 16C13.9 16 13 15.1 13 14C13 13.82 13.02 13.64 13.07 13.48L10.52 10.93C10.36 10.98 10.18 11 10 11C9.82 11 9.64 10.98 9.48 10.93L4.93 15.49C4.98 15.65 5 15.82 5 16C5 17.1 4.1 18 3 18C1.9 18 1 17.1 1 16C1 14.9 1.9 14 3 14C3.18 14 3.35 14.02 3.51 14.07L8.07 9.52C8.02 9.36 8 9.18 8 9C8 7.9 8.9 7 10 7C11.1 7 12 7.9 12 9C12 9.18 11.98 9.36 11.93 9.52L14.48 12.07C14.64 12.02 14.82 12 15 12C15.18 12 15.36 12.02 15.52 12.07L19.07 8.51C19.02 8.35 19 8.18 19 8C19 6.9 19.9 6 21 6C22.1 6 23 6.9 23 8Z" fill="currentColor"></path>
  </svg>
</ic-navigation-item>
`;

exports[`ic-badge should render slotted in a tab: should render slotted in a tab 1`] = `
<ic-tab aria-disabled="false" aria-selected="false" context-id="default" role="tab" tabindex="-1">
  <mock:shadow-root>
    <ic-typography class="ic-tab-label" variant="label">
      <span>
        <slot></slot>
      </span>
    </ic-typography>
    <slot name="badge"></slot>
  </mock:shadow-root>
  Tab
  <ic-badge aria-label="badge being displayed" class="default foreground-light inline neutral show text" position="inline" role="status" slot="badge" text-label="1">
    <mock:shadow-root>
      <ic-typography variant="badge">
        1
      </ic-typography>
    </mock:shadow-root>
  </ic-badge>
</ic-tab>
`;

exports[`ic-badge should render success variant: should render success variant 1`] = `
<ic-button class="button-size-default button-variant-primary with-badge" exportparts="button">
  <mock:shadow-root>
    <slot name="badge"></slot>
    <button class="button" part="button" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
  <ic-badge aria-label="badge being displayed" class="default far foreground-light show success text" role="status" slot="badge" text-label="1" variant="success">
    <mock:shadow-root>
      <ic-typography variant="badge">
        1
      </ic-typography>
    </mock:shadow-root>
  </ic-badge>
</ic-button>
`;

exports[`ic-badge should render type dot: should render type dot 1`] = `
<ic-button class="button-size-default button-variant-primary with-badge" exportparts="button">
  <mock:shadow-root>
    <slot name="badge"></slot>
    <button class="button" part="button" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
  <ic-badge aria-label="badge being displayed" class="default dot far foreground-light neutral show" role="status" slot="badge" type="dot">
    <mock:shadow-root></mock:shadow-root>
  </ic-badge>
</ic-button>
`;

exports[`ic-badge should render type icon: should render type icon 1`] = `
<ic-button class="button-size-default button-variant-primary with-badge" exportparts="button">
  <mock:shadow-root>
    <slot name="badge"></slot>
    <button class="button" part="button" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
  <ic-badge aria-label="badge being displayed" class="default far foreground-light icon neutral show" role="status" slot="badge" type="icon">
    <mock:shadow-root>
      <slot name="badge-icon"></slot>
    </mock:shadow-root>
    <svg fill="#000000" height="24px" slot="badge-icon" viewBox="0 0 24 24" width="24px" xmlns="http://www.w3.org/2000/svg">
      <path d="M0 0h24v24H0V0z" fill="none"></path>
      <path d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"></path>
    </svg>
  </ic-badge>
</ic-button>
`;

exports[`ic-badge should render warning variant: should render warning variant 1`] = `
<ic-button class="button-size-default button-variant-primary with-badge" exportparts="button">
  <mock:shadow-root>
    <slot name="badge"></slot>
    <button class="button" part="button" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
  <ic-badge aria-label="badge being displayed" class="default far foreground-light show text warning" role="status" slot="badge" text-label="1" variant="warning">
    <mock:shadow-root>
      <ic-typography variant="badge">
        1
      </ic-typography>
    </mock:shadow-root>
  </ic-badge>
</ic-button>
`;

exports[`ic-badge should render with accessible label: should render with accessible label 1`] = `
<ic-button class="button-size-default button-variant-primary with-badge" exportparts="button">
  <mock:shadow-root>
    <slot name="badge"></slot>
    <button class="button" part="button" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
  <ic-badge accessible-label="1 notification found" aria-label="badge displaying 1 notification found" class="default far foreground-light neutral show text" role="status" slot="badge" text-label="1">
    <mock:shadow-root>
      <ic-typography variant="badge">
        1
      </ic-typography>
    </mock:shadow-root>
  </ic-badge>
</ic-button>
`;

exports[`ic-badge should render with id set: should render with id 1`] = `
<ic-button class="button-size-default button-variant-primary with-badge" exportparts="button">
  <mock:shadow-root>
    <slot name="badge"></slot>
    <button class="button" part="button" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
  <ic-badge aria-label="badge being displayed" class="default far foreground-light neutral show text" id="badge-1" role="status" slot="badge" text-label="1">
    <mock:shadow-root>
      <ic-typography variant="badge">
        1
      </ic-typography>
    </mock:shadow-root>
  </ic-badge>
</ic-button>
`;

exports[`ic-badge should render with max number prop set: should render with max number prop set 1`] = `
<ic-button class="button-size-default button-variant-primary with-badge" exportparts="button">
  <mock:shadow-root>
    <slot name="badge"></slot>
    <button class="button" part="button" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
  <ic-badge aria-label="badge being displayed" class="default far foreground-light neutral show text" max-number="9" role="status" slot="badge" text-label="100">
    <mock:shadow-root>
      <ic-typography variant="badge">
        9+
      </ic-typography>
    </mock:shadow-root>
  </ic-badge>
</ic-button>
`;

exports[`ic-badge should render with text slotted in a button: should render with text slotted in a button 1`] = `
<ic-button class="button-size-default button-variant-primary with-badge" exportparts="button">
  <mock:shadow-root>
    <slot name="badge"></slot>
    <button class="button" part="button" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
  <ic-badge aria-label="badge being displayed" class="default far foreground-light neutral show text" role="status" slot="badge" text-label="1">
    <mock:shadow-root>
      <ic-typography variant="badge">
        1
      </ic-typography>
    </mock:shadow-root>
  </ic-badge>
</ic-button>
`;

exports[`ic-badge should render with visible false: should render with visible false 1`] = `
<ic-button class="button-size-default button-variant-primary with-badge" exportparts="button">
  <mock:shadow-root>
    <slot name="badge"></slot>
    <button class="button" part="button" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
  <ic-badge aria-label="badge being displayed" class="default far foreground-light hide neutral text" role="status" slot="badge" text-label="1" visible="false">
    <mock:shadow-root>
      <ic-typography variant="badge">
        1
      </ic-typography>
    </mock:shadow-root>
  </ic-badge>
</ic-button>
`;
